### ISSUE TITLE
UKHPI landing page fails and displays 'major error' indication

### ISSUE THREAD

**Issue by bwmcbride** [epimorphics/ukhpi#22]
This may be due to the expected data not being in the store at the time of the access.

In general, this should not happen, but if the page happens to be unable to retrieve the current index data would be better to display the rest of the page.

**Comment by ijdickinson** [comment]
I've added some exception handling so that the page does render, albeit not that prettily, if the dsapi is down

**Comment by ijdickinson** [comment]
Are there any other edge cases we should handle (that I can easily test)?

**Comment by bwmcbride** [comment]
Confirmed that if dsapi is down, then the page renders but with 'unknown' current data.

Still get the major error indicator if the data server is unreachable.

**Comment by ijdickinson** [comment]
I've added robustness against the errors arising from the DsAPI being unable to contact the remote data server or getting a 404 from the remote end. We are planning to run further tests to see what happens if the Fuseki service is dead or unresponsive, but that requires more setup.

**Comment by ijdickinson** [comment]
Brian and I conducted some further tests. If the remote Fuseki process is dead, DsAPI incorrectly reports a 400 error, which wasn't being handled gracefully. Now fixed. 

If the remote Fuseki is massively overloaded, the query queues up and returns eventually. This makes e.g. the landing page slow to load, but doesn't cause a failure. We might want to look at putting a timeout on the landing page query as a UX improvement, but I believe that we can now close this ticket as we don't see any more major failures under the scenarios we have tested.
